chain/src/pipe.rs

```diff
@@ -148,7 +149,13 @@ pub fn process_block(
 		// First we rewind the txhashset extension if necessary
 		// to put it into a consistent state for validating the block.
 		// We can skip this step if the previous header is the latest header we saw.
-		if is_next_block(&b.header, ctx) {
+
+		//
+		// TODO - Is this safe to do? Can we just go look in the db via the batch?
+		//
+		let head = extension.batch.head()?;
```

Comment:
Yes, I think it's fine now that the lock is taken earlier. What matter the most is that another can't update the head without us knowing because we're in different batches.